src/components/DevelopmentGuide/ConditionalViewport.mdx

```diff
@@ -0,0 +1,111 @@
+
+The viewport is the visible area of a web page, which varies from device to device.
+The user interacts with the viewport by scrolling or zooming in and out. The viewport's dimensions and scale can be set from the `<meta>` tag.
+
+On iOS devices, non-native websites will automatically zoom in when a user selects an input element (changing the scale of the viewport). This default is meant to ensure any input field is accessible for the visually impaired, but it generally forces users to zoom back out to continue viewing the page.
+
+This auto-zoom does not occur on Android mobile devices, so **it requires a device-specific solution to be implemented by the consuming app** (rather than, for example, changing the DS `TextInput` component to resist the zoom behavior).
+
+## How to resolve
+
+The general solution to prevent auto-zoom on iOS devices is to conditionally set the `<meta>` viewport tag.
+
+For iOS:
+
+<Source
+  code={`
+<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
+`}
+  language="html"
+/>
+
+The `maximum-scale` value prevents the auto-zoom from occurring on iOS, but when
+applied on Android, it prevents users from zooming in correctly at all (which is
+its own accessibility violation).
+
+So for everything else, the viewport tag should be:
+
+<Source
+  code={`
+<meta name="viewport" content="width=device-width">
+`}
+  language="html"
+/>
+
+Setting the viewport conditionally requires capturing whether the device is iOS or not.
+How you do this will depend on your app– the below approaches are for Next.js apps (app or pages router).
+
+### Approach for Next.js App router apps
+
+Using the App router, you can customize the initial viewport with the dynamic [`generateViewport()`](https://nextjs.org/docs/app/api-reference/functions/generate-viewport) function. This only works in a server component, so most likely will be defined near the root of your project in
+the `page.tsx` or `layout.tsx` file.
+
+Then, you can access the [`userAgent`](https://nextjs.org/docs/app/api-reference/functions/userAgent) object from Next's [`headers()`](https://nextjs.org/docs/app/api-reference/functions/headers) function. If the current device is an iPhone, return the necessary viewport object.
+
+<Source
+  code={`
+
+// layout.tsx or page.tsx
+
+export async function generateViewport() {
+  const userAgent = headers().get("user-agent");
+  const isiPhone = /iphone/i.test(userAgent ?? "");
+  return isiPhone ? 
+  { width: "device-width", initialScale: 1, maximumScale: 1, userScalable: false }
+    : {};
+}
+    
+`}
+language="js"
+/>
+
+This approach is currently implemented on [Digital Collections](https://github.com/NYPL/digital-collections/blob/8758f1106af59c4a76015d978700ec8baba6deb4/app/layout.tsx#L45).
+
+### Approach for Next.js Pages router apps
+
+In a Pages router app, you need to find the `userAgent` object from a client component (again, probably near the root of your project).
```

Comment:
Have you had a chance to try this in Turbine? If so, you can use a branch as a reference similar to how you linked DC above.

If no server-side approach exists, mention that. I'd prefer a server-side approach over client-side but it's not a big deal. If there is one, don't remove this existing example. It's possible some apps may not be able to implement the server-side approach.

Reply:
No serverside approach since Next does not allow either `_app` and `_document` to use `getStaticProps` or `getServerSideProps` and I think a hook is necessary here to get the `userAgent` object. Getting the viewport update merged into Turbine (of the approach described here) and will add a link